packages/odata-v4/src/de-serializers/converters.ts

```diff
@@ -42,8 +42,25 @@ export function deserializeDateTimeOffsetToMoment(
 /**
  * @internal
  */
-export function serializeToDateTimeOffset(value: moment.Moment): string {
-  return value.utc().format('YYYY-MM-DDTHH:mm:ss.SSS') + 'Z';
+export function serializeToDateTimeOffset(
+  value: moment.Moment,
+  precision?: number
+): string {
+  // For temporal values, OData specification restricts precision values to a non-negative integer between 0 and 12
+  if (precision !== undefined && !validatePrecision(precision)) {
+    throw new Error(
+      `Provided precision value: ${precision} is invalid. Precision must lie between 0 and 12`
+    );
+  }
```

Comment:
@jjtang1985 We also decided to throw an error here (different from previous behaviour) as OData specification only permits precision be a non-negative integer between zero and twelve.

CAP seems to restrict the precision for Timestamp CDS(`Edm.DateTimeOffset` with Precision="7") type too, but we are are only enforcing what OData spec mentions.

Please let me know, if this is okay.